### DISCUSSION TITLE
Naming branches

### DISCUSSION THREAD

**Discussion by rnugent3** [HydrologicEngineeringCenter/HEC-FDA#92] [General]
@HenryGeorgist I see that there are many different views on the best practices for naming branches. Would you mind expressing your view or point to an article with which you are agreeable?

**Comment by HenryGeorgist** [comment]
I dont feel super strongly.  a common practice is something like this:

feature/<brief description> (e.g. Feature/LP3Tests, or Feature/LP3Tests#63)
bug/<brief description>

**Comment by Brennan1994** [comment]
Seems to me like this is a low consequence choice. If we're sticking to our guns on short branch life, none of these will last long, and we'll never see their names again once they're merged. That said, I'll conform to whatever. Just so long as it's descriptive of the task being done.

**Reply by HenryGeorgist** [reply]
The branch names can be used to identify features via an api query on the project - so if we were really good about our management here, we could rely on our naming convention to generate a feature list automatically for each new release.

**Reply by Brennan1994** [reply]
Oooh. I like that. That's cool.

**Comment by rnugent3** [comment]
I’m hearing little downside but opportunities for upside. We could use branch names as another device for demonstrating our use of SET funds?

**Comment by rnugent3** [comment]
We've not been very regular on naming branches. I think this is OK for FY22 because we're moving very fast. In FY23, I hope that we can tighten this up to help us with reporting.
